lib/vrl/compiler/src/test_util.rs

```diff
@@ -104,7 +104,7 @@ macro_rules! test_function {
                         let got_value = expression.resolve(&mut ctx)
                             .map_err(|e| format!("{:#}", anyhow::anyhow!(e)));
 
-                        assert_eq!(got_value, want);
+                        assert!(got_value == want, "assertion failed for `{}` case:\n  got:    {:?}\n  wanted: {:?}", stringify!($case), got_value, want);
```

Comment:
I changed this because otherwise the line location of the panic (when the assertion failed) is always the line where the `test_function!` macro is called, not the actual `assert_eq!` that failed, so you had no clue which test case was failing.